src/main/groovy/edu/oregonstate/mist/locations/LocationUtil.groovy

```diff
@@ -0,0 +1,99 @@
+package edu.oregonstate.mist.locations
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.security.MessageDigest
+
+class LocationUtil {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocationUtil.class)
+
+    public static final String VALID_LAT_LONG = "-?\\d+(\\.\\d+)?"
+
+    final Map<String, String> locationConfiguration
+
+    /**
+     * Working directory where the cache and downloaded / generated files are stored.
+     */
+    private final String cacheDirectory
+
+    LocationUtil(Map<String, String> locationConfiguration) {
+        this.locationConfiguration = locationConfiguration
+        cacheDirectory = locationConfiguration.get("cacheDirectory")
+    }
+
+    /**
+     * Calculates MD5 Hash of a string
+     *
+     * @param content
+     * @return
+     */
+    public static String getMD5Hash(String content) {
+        MessageDigest.getInstance("MD5").digest(content.bytes).encodeHex().toString()
+    }
+
+    /**
+     * Tries to get the data from the URL. If fetching from the url fails, it retrieves the
+     * data from the cache directory
+     *
+     * @param URL
+     * @param cachedFile
+     * @return
+     * @throws Exception
+     */
+    public String getDataFromUrlOrCache(String URL, String cachedFile) throws Exception {
```

Comment:
This method does a web request and calls another method to cache information to the local filesystem. There's not just a single type of exception that could be thrown since there could be a file permission error, not being able to find the file or cache directory, a 404 error thrown when trying to access the web (this actually throws an exception), a networking issue that prevents fetching the data.